api/src/main/java/org/pentaho/platform/api/genericfile/IGenericFileService.java

```diff
@@ -275,4 +276,14 @@ default IGenericFile getFile( @NonNull String path ) throws OperationFailedExcep
    */
   @NonNull
   IGenericFile getFile( @NonNull GenericFilePath path ) throws OperationFailedException;
+
+  /**
+   * Gets a list of deleted files.
```

Comment:
Maybe, something like: `Gets a list of deleted files which are still available in the trash folder.` ?
(if you change this, sync with the provider)